scripts/create_dataset.py

```diff
@@ -46,26 +46,323 @@ def init_dataset(dataset_name, dataset_description):
+            with open(os.path.join(parent_directory, file), "r") as json_file:
                 json_as_dict = json.load(json_file)
-            dataset["schema"][schema_type].append(json_as_dict)
+            # add the name field field of the json as a kay to make each json searchable
+            input_dict.update({json_as_dict["name"]: json_as_dict})
+
+    return input_dict
+
+
+def create_and_add_tables_and_relationships(input_tables, dataset):
+    """
+    Creates tables and relationships and then adds them to the dataset's relationship and asset fields.
+    :param input_tables: the input json tables that have been added to a dict
+    :param dataset: the created dataset that will be written to disk
+    """
+    for input_table_name in input_tables:
+        table, relationships = create_table_and_relationships(input_table=input_tables[input_table_name],
+                                                              input_tables=input_tables)
+        dataset["schema"]["tables"].append(table)
+
+        # only add a relationship the schema if any of the column from the given input table contains reference(s)
+        if len(relationships) > 0:
+            dataset["schema"]["relationships"].extend(relationships)
+
+    return dataset
+
+
+def create_and_add_assets(input_assets, dataset):
+    """
+    For every input asset it creates an asset by adding all the tables and relationship names to the asset and
+    then adds each created assets to the dataset.
+    :param input_assets: the input json assets that have been added to a dict
+    :param dataset: the created dataset that will be written to disk
+    """
+    for input_asset_name in input_assets:
+        asset = create_asset(input_asset=input_assets[input_asset_name],
+                             dataset=dataset)
+        dataset["schema"]["assets"].append(asset)
+
     return dataset
 
 
+def create_asset(input_asset, dataset):
+    """
+    Creates an asset by adding all the tables and relationship names to the asset.
+    :param input_asset: the input json asset that has been added to a dict
+    :param dataset: the created dataset that will be written to disk
+    """
+    asset = init_asset(asset_name=input_asset["name"],
+                       root_table=input_asset["rootTable"],
+                       root_column=input_asset["rootColumn"])
+
+    # add tables and the follow section to the asset
+    asset_with_tables = add_asset_tables(asset=asset, dataset=dataset)
+    asset_with_tables_and_follow = add_asset_follow(asset=asset_with_tables, dataset=dataset)
+
+    return asset_with_tables_and_follow
+
+
+def add_asset_tables(asset, dataset):
+    """
+    Adds all the table names from the dataset to the asset.
+    :param asset: the asset to be created and added to the dataset with table names
+    :param dataset: the created dataset that will be written to disk
+    """
+    for table in dataset["schema"]["tables"]:
+        asset["tables"].append(
```

Comment:
Same point as above: it'd be cleaner not to mutate. This method could return the "asset form" of the list of tables given to it, and the caller can then use that list to initialize asset definitions.